libnmstate/compat.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+def subnet_of(a, b):
```

Comment:
I think the way it should go is:
```
import ipaddress
import iplib

if six.PY2:
    ipaddress.IPv4Network.subnet_of = iplib.subnet_of
```

But I am missing something here.
`subnet_of` exists in the python-ipaddress package (or directly from pip) which is a backport of py3 ipaddress.
If we depend on that one, we will not need any hack.
Or do I miss something.

Reply:
The `subnet_of()` is new in version python 3.7, above line is for python 3.6-.

Reply:
> the user should import ipaddress and use it the same way it is documented in the docs.

Who is the `user` here? It's not a API, for internal use, internal function call should be through `iplib.is_subnet_of` which hide the detail of using `ipaddress`.

Reply:
The user is anyone, internal or external. There is no difference between internal and external API except for the flexibility to change the internal one more frequently.

A developer should not think about this stuff at all. If this is already introduced in 3.7 under ipaddress, we can make it look as if all versions have this available and when we reach 3.7 only support, we can drop the patching/wrappers.